File: 2733-Neither-Minimum-nor-Maximum/n_solution.py
# https://leetcode.com/problems/neither-minimum-nor-maximum/submissions/1027042098/

class Solution:
    def findNonMinOrMax(self, nums: List[int]) -> int:
        max_num = None
        min_num = None
        
        for num in nums:
            if max_num is None or num > max_num:
                max_num = num
            if min_num is None or num < min_num:
                min_num = num                   

        neither = -1
        for num in nums:
            if num != max_num and num != min_num:
                neither = num

        return neither

File: 2485-Find-the-Pivot-Integer/n_solution.ts
// https://leetcode.com/problems/find-the-pivot-integer/submissions/1202087031/

function pivotInteger(n: number): number {
    const nums = new Map();
    let sum = 0;
    for(let i = n; i > 0; --i) {
        sum = i + sum;
        nums.set(i, sum);
    }

    sum = 0;
    for(let i = 1; i <= n; ++i){
        sum = i + sum;
        if(nums.get(i) === sum) return i
    }
    return -1;
};
